src/sparql/exact_syn_from_label.ru

```diff
@@ -1,11 +1,29 @@
-# Create an exact synonym copy for all labels
 PREFIX oboInOwl: <http://www.geneontology.org/formats/oboInOwl#>
 PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
 PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
 
+# Create an exact synonym copy for all labels using the "fixed" label that has brackets
+
 INSERT {
-  ?cls oboInOwl:hasExactSynonym ?label .
-} WHERE {
+  <http://purl.obolibrary.org/obo/mondo#GENERATED_FROM_LABEL> rdf:type owl:AnnotationProperty .
+  <http://www.geneontology.org/formats/oboInOwl#hasSynonymType> rdf:type owl:AnnotationProperty .
+  <http://purl.obolibrary.org/obo/mondo#GENERATED_FROM_LABEL> rdfs:subPropertyOf <http://www.geneontology.org/formats/oboInOwl#SynonymTypeProperty>  .
+
+  ?cls oboInOwl:hasExactSynonym ?cleanLabel .
+
+  [
+    rdf:type owl:Axiom ;
+    owl:annotatedSource ?cls ;
+    owl:annotatedProperty oboInOwl:hasExactSynonym ;
+    owl:annotatedTarget ?cleanLabel ;
+    oboInOwl:hasSynonymType <http://purl.obolibrary.org/obo/mondo#GENERATED_FROM_LABEL>
+  ] .
+} 
+WHERE {
   ?cls rdfs:label ?label .
+
+  BIND(REPLACE(STR(?label), " [\\[(].*[)\\]]$", "") AS ?cleanLabel)
```

Comment:
I tested this update with ICD11 and here's an example IRI, http://id.who.int/icd/entity/1945176477, where it's using the "cleaned" label to convert to the exact synonym tagged as `GENERATED_FROM_LABEL`. Tested by running: `sh run.sh make components/icd11foundation.owl -B`

Reply:
@matentzn are you ok with this addition to the query or do you prefer the original label be used as the `GENERATED_FROM_LABEL` exact synonym before any attempts to "fix" the label due to the presence of brackets? 

One question I have is how well this "fix" for labels was previously tested and reviewed to confirm it's working as expected in all cases so that we do not add badly formatted synonyms into Mondo.

Reply:
I am not sure. This fix was needed for ICD10, and was not really tested with anything else. I am not sure this is a good idea to make default for _all_ ontologies. I would do this only for ICD11 and ICD10. There are some legitimate reasons of having brackets in class names..

Reply:
Maybe I would just leave the unchanged label for now.. for safety reasons

Reply:
Sure, while I would like to have more meaningful synonyms, ie without brackets etc., if this bracket "fix" was not more deeply tested then we run the risk of adding even worse formatted synonyms into Mondo which is a big negative. I will revert this change, but this small though exercise has been useful.